lib/jack.jl

```diff
@@ -0,0 +1,33 @@
+using JLLPrefixes
+
+# Helper to extract buildkite environment arrays
+function extract_env_array(prefix::String)
+    envname(idx::Int) = string(prefix, "_", idx)
+
+    idx = 0
+    array = String[]
+    while haskey(ENV, envname(idx))
+        push!(array, ENV[envname(idx)])
+        idx += 1
+    end
+    return array
+end
```

Comment:
This is good, but you need to check not only for `BUILDKITE_PLUGIN_JACK_INSTALL_${idx}`, but also `BUILDKITE_PLUGIN_JACK_INSTALL_${idx}_NAME`, since you have the two different syntaxes:

```
            - "FFMPEG_jll"
            - name: "GNUMake_jll"
              version: "v4.4.0+0"
```

Reply:
For debugging these kinds of things, I like to do this at the top of my plugin hook:

```
echo "--- DEBUGGING ENV DUMP"
env | grep BUILDKITE_PLUGIN_JACK
```

That gives me a nice dump of all the environment variables that I might care about in the log.

Reply:
> For debugging these kinds of things, I like to do this at the top of my plugin hook:
> ```
> echo "--- DEBUGGING ENV DUMP"
> env | grep BUILDKITE_PLUGIN_JACK
> ```
> That gives me a nice dump of all the environment variables that I might care about in the log.

Ah, this is nifty. Saves me some time that would've been spent figuring out how other plugins do it otherwise. 😄